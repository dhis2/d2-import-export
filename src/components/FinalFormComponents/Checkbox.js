import { useField } from 'react-final-form'
import { Checkbox as CoreCheckbox } from '@dhis2/ui-core'
import React from 'react'
import propTypes from 'prop-types'

export const Checkbox = ({ label, name, checkedInitially: initialValue }) => {
    const { input } = useField(name, {
        initialValue,
        type: 'checkbox',
        format: v => v.toString(),
        parse: v => Boolean(v),
    })

    return (
        <CoreCheckbox
            {...input}
            checked={input.value === 'true'}
            label={label}
        />
    )
}

Checkbox.propTypes = {
    label: propTypes.string.isRequired,
    name: propTypes.string.isRequired,
    checkedInitially: propTypes.bool,
}
