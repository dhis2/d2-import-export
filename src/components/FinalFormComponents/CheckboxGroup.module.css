.actions {
    margin-bottom: 4px;
}

.selectAll {
    margin-right: 4px;
}

.options {
    max-height: 250px;
    overflow-y: scroll;
}
