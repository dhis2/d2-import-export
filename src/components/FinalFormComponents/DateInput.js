import { useField } from 'react-final-form'
import DatePicker from 'material-ui/DatePicker'
import React from 'react'
import propTypes from 'prop-types'

import styles from './DateInput.module.css'

export const DateInput = ({ minDate, name, initialValue }) => {
    const { input } = useField(name, {
        type: 'input',
        initialValue,
        format: value => new Date(value),
    })

    const onChange = (_, date) => input.onChange(date)

    return (
        <div>
            <DatePicker
                className={styles.datePicker}
                textFieldStyle={{ height: 36, fontSize: 15, fontWeight: 500 }}
                autoOk={true}
                key={name}
                name={name}
                minDate={minDate}
                value={input.value}
                onChange={onChange}
            />
        </div>
    )
}

DateInput.propTypes = {
    name: propTypes.string.isRequired,
    format: propTypes.func,
    initialValue: propTypes.instanceOf(Date),
    minDate: propTypes.string,
}

DateInput.defaultProps = {
    initialValue: new Date(),
}
