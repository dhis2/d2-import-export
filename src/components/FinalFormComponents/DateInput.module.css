.datePicker {
    height: 36px;
}
