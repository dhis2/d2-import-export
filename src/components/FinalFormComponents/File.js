import { useField } from 'react-final-form'
import { Help } from '@dhis2/ui-core'
import React, { useRef, useCallback } from 'react'
import propTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import styles from './File.module.css'

const FileUploadIcon = () => (
    <svg className={styles.svg}>
        <path d="M0 0h24v24H0z" fill="none" />
        <path d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z" />
    </svg>
)

export const File = ({ name, dataTest }) => {
    const ref = useRef()
    const { input, meta } = useField(name, {
        type: 'file',
        parse: () => ref.current.files[0],
        validate: value => (!!value ? undefined : i18n.t('Required')),
    })
    const onClick = useCallback(() => {
        ref.current.click()
    }, [ref])

    const { value, ...withoutValue } = input
    const label = value ? value.name : i18n.t('Choose a file to upload') + ' *'

    return (
        <div className={styles.container} data-test={dataTest}>
            <input {...withoutValue} ref={ref} className={styles.input} />

            <button type="button" onClick={onClick} className={styles.button}>
                <FileUploadIcon />
                <span className={styles.label}>{label}</span>
            </button>

            {meta.touched && meta.error && (
                <div className={styles.error}>
                    <Help error>{meta.error}</Help>
                </div>
            )}
        </div>
    )
}

File.propTypes = {
    name: propTypes.string,
}
