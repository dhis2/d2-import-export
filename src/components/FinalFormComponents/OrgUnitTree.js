import { useDispatch, useSelector } from 'react-redux'
import { useField } from 'react-final-form'
import React, { useEffect } from 'react'
import propTypes from 'prop-types'

import { Tree } from '../Tree'
import { closeOrgUnit } from '../../reducers/orgUnits/actions'
import {
    getOrgUnitRootLoaded,
    getOrgUnitRootLoading,
    getOrgUnits,
} from '../../reducers/orgUnits/selectors'
import { openPath, loadRootOrgUnit } from '../../reducers/orgUnits/thunks'

export const OrgUnitTree = ({ name, selected, multiple }) => {
    const dispatch = useDispatch()
    const list = useSelector(getOrgUnits)
    const rootLoaded = useSelector(getOrgUnitRootLoaded)
    const rootLoading = useSelector(getOrgUnitRootLoading)
    const onIconClick = (path, open) => {
        if (open) {
            dispatch(openPath(path))
        } else {
            dispatch(closeOrgUnit(path))
        }
    }

    useEffect(() => {
        if (!rootLoaded && !rootLoading) {
            dispatch(loadRootOrgUnit())
        }
    }, [rootLoaded, rootLoading, dispatch])

    const { input } = useField(name, {
        initialValue: selected,
        type: 'select',
    })

    return (
        <Tree
            multiple={multiple}
            selectable={true}
            list={list}
            selected={input.value}
            onIconClick={onIconClick}
            setSelected={input.onChange}
        />
    )
}

OrgUnitTree.propTypes = {
    name: propTypes.string.isRequired,
    selected: propTypes.arrayOf(propTypes.string).isRequired,
    multiple: propTypes.bool,
}

OrgUnitTree.defaultProps = {
    selected: [],
    multiple: true,
}
