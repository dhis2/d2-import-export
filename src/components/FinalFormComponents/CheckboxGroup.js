import { Button } from '@dhis2/ui-core'
import { useForm } from 'react-final-form'
import React from 'react'
import propTypes from 'prop-types'

import { Checkbox } from './Checkbox'
import styles from './CheckboxGroup.module.css'

export const CheckboxGroup = ({ name, options }) => {
    const form = useForm()

    const selectAll = () =>
        form.change(
            name,
            options.map(({ value }) => value)
        )
    const clearAll = () => form.change(name, [])

    return (
        <div>
            <div className={styles.actions}>
                <Button className={styles.selectAll} onClick={selectAll}>
                    Select all
                </Button>
                <Button onClick={clearAll}>Clear all</Button>
            </div>

            <div className={styles.options}>
                {options.map(({ label, value }) => (
                    <Checkbox
                        key={label}
                        name={name}
                        label={label}
                        value={value}
                        multiple
                    />
                ))}
            </div>
        </div>
    )
}

CheckboxGroup.propTypes = {
    name: propTypes.string.isRequired,
    options: propTypes.arrayOf(
        propTypes.shape({
            label: propTypes.string.isRequired,
            value: propTypes.string.isRequired,
        })
    ).isRequired,
}
