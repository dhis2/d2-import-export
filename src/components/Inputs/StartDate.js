import React from 'react'
import i18n from '@dhis2/d2-i18n'

import { DateInput } from '../FinalFormComponents/DateInput'
import { Field } from '../Field/Field'
import { Label } from '../Field/Label'

const createDefaultValue = () => {
    const date = new Date()
    const month = date.getMonth()

    date.setMonth(month - 3)

    return date
}

export const START_DATE_KEY = 'startDate'
export const START_DATE_DEFAULT_VALUE = createDefaultValue()

export const StartDate = () => (
    <Field>
        <Label>{i18n.t('StartDate')}</Label>
        <DateInput name={START_DATE_KEY} label={i18n.t('Start date')} />
    </Field>
)
