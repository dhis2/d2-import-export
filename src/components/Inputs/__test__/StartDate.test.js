import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { DateInput } from '../../FinalFormComponents/DateInput'
import { StartDate } from '../StartDate'

jest.mock('react-final-form')

describe('Input component - StartDate', () => {
    const dateToUse = new Date('2020')
    DateInput.defaultProps.initialValue = dateToUse

    it('should render correctly', () => {
        const file = shallow(<StartDate />)
        expect(toJson(file)).toMatchSnapshot()
    })
})
