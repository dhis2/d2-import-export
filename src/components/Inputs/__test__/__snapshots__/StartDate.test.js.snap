// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - StartDate should render correctly 1`] = `
<Field>
  <Label>
    StartDate
  </Label>
  <DateInput
    initialValue={2020-01-01T00:00:00.000Z}
    label="Start date"
    name="startDate"
  />
</Field>
`;
